photon-ml/src/main/scala/com/linkedin/photon/ml/FeatureIndexingJob.scala

```diff
@@ -107,16 +132,24 @@ class FeatureIndexingJob(val sc: SparkContext,
     keyedFeaturesUnionedRDD.distinct().groupByKey(new HashPartitioner(partitionNum))
```

Comment:
@timyitong Can you remind me why the fact that "integer's hashcode is still itself" would save the amount of data to be shuffled?